File: packages/js/core/src/uri-resolvers/redirects/RedirectsResolver.ts
import { applyRedirects } from "../../../algorithms";
import { Uri, Client } from "../../../types";
import { UriResolver, UriResolutionResult } from "../../core";

export class RedirectsResolver implements UriResolver {
  public get name(): string {
    return RedirectsResolver.name;
  }

  async resolveUri(uri: Uri, client: Client): Promise<UriResolutionResult> {
    const redirectedUri = applyRedirects(uri, client.getRedirects({}));

    return Promise.resolve({
      uri: redirectedUri,
    });
  }
}

File: packages/py/core/core/uri_resolution/resolvers/redirects/redirects_resolver.py
from __future__ import annotations

from typing import Awaitable

from ....algorithms import apply_redirects
from ....types import Client, GetRedirectsOptions, Uri, WrapperCache
from ...core import UriResolutionResult, UriResolutionStack, UriResolver


class RedirectsResolver(UriResolver):
    @property
    def name(self) -> str:
        return type(self).__name__

    async def resolve_uri(
        self,
        uri: Uri,
        client: Client,
        cache: WrapperCache,
        resolution_path: UriResolutionStack,
    ) -> Awaitable(UriResolutionResult):
        redirect_uri = apply_redirects(
            uri, client.get_redirects(GetRedirectsOptions(""))
        )

        return UriResolutionResult(uri=redirect_uri)
